src/Conda.jl

```diff
@@ -34,43 +34,95 @@ module Conda
 using Compat
 import Compat.String
 using JSON
+include("../deps/deps.jl")
 
-"Prefix for installation of all the packages."
-const PREFIX = abspath(dirname(@__FILE__), "..", "deps", "usr")
+if !isdir(default_dir)
+    # Ensure default_dir exists
+    mkpath(default_dir)
+end
+
+type Environment
+    path::ASCIIString
+    function Environment(path::String)
+        if (!isdir(path))
+            return error("Path to conda environment is not valid.")
+        end
+        return new(path)
+    end
+    function Environment(name::Symbol)
+        if (length(string(name)) == 0)
+            return error("Environment name should be non empty.")
+        end
+        return new(joinpath(default_dir, "envs", string(name)))
+    end
+end
+
+RootEnv = Environment(default_dir)
+
+"Prefix for installation of the environment"
+function prefix(env::Environment)
+    return env.path
+end
+const PREFIX = prefix(RootEnv)
 
 "Prefix for the executable files installed with the packages"
-const BINDIR = is_windows() ? joinpath(PREFIX, "Library", "bin") : joinpath(PREFIX, "bin")
+function bin_dir(env::Environment)
+    return is_windows() ? joinpath(prefix(env), "Library", "bin") : joinpath(prefix(env), "bin")
+end
+const BINDIR = bin_dir(RootEnv)
+
+"Prefix for the shared libraries installed with the packages"
+function lib_dir(env::Environment)
+    return is_windows() ? joinpath(prefix(env), "Library", "bin") : joinpath(prefix(env), "lib")
+end
+const LIBDIR = lib_dir(RootEnv)
 
 "Prefix for the python scripts. On UNIX, this is the same than Conda.BINDIR"
-const SCRIPTDIR = is_windows() ? joinpath(PREFIX, "Scripts") : BINDIR
+function script_dir(env::Environment)
+    return is_windows() ? joinpath(prefix(env), "Scripts") : bin_dir(env)
+end
+const SCRIPTDIR = script_dir(RootEnv)
 
 "Prefix where the `python` command lives"
-const PYTHONDIR = is_windows() ? PREFIX : BINDIR
+function python_dir(env::Environment)
+    return is_windows() ? prefix(env) : bin_dir(env)
+end
+const PYTHONDIR = python_dir(RootEnv)
 
-"Path to the `conda` binary"
-const conda = joinpath(SCRIPTDIR, "conda")
+function conda_bin(env::Environment)
+    if is_windows()
+        p = script_dir(env)
+        return isfile(joinpath(p, "conda.bat")) ? joinpath(p, "conda.bat") : joinpath(p, "conda.exe")
+    else
+        return joinpath(script_dir(env), "conda")
+    end
+end
+const conda = conda_bin(RootEnv)
 
 "Path to the condarc file"
-const CONDARC = joinpath(PREFIX, "condarc-julia")
+const CONDARC = joinpath(prefix(RootEnv), "condarc-julia")
+
 
 """
 Use a cleaned up environment for the command `cmd`.
 
 Any environment variable starting by CONDA will interact with the run.
 """
-function _set_conda_env(cmd)
-    env = copy(ENV)
+function _set_conda_env(cmd, env::Environment=RootEnv)
+    env_hash = copy(ENV)
```

Comment:
This name is misleading (at least to me =)). I Though this was the result of `hash(env)`. How about `env_var` or `variables` or `env_variable`?